src/main/scala/com/horizon/exchangeapi/PatternsRoutes.scala

```diff
@@ -301,7 +302,7 @@ trait PatternRoutes extends ScalatraBase with FutureSupport with SwaggerSupport
   post("/orgs/:orgid/patterns/:pattern", operation(postPatterns)) ({
     val orgid = params("orgid")
     val barePattern = params("pattern")   // but do not have a hack/fix for the name
-    val pattern = OrgAndId(orgid,barePattern).toString
+    val pattern = OrgAndId(orgid,barePattern).toString.trim
```

Comment:
With doing trim on the result of OrgAndId, couldn't spaces still remain if they were at the **beginning** of pattern? Also, i guess theoretically orgid could have spaces too. How about putting the trims inside OrgAndId()? That would take care of both of these problems and also handle all of the other type of id's (nodes, etc.). Also add the trim to OrgAndIdCred().